src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx

```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
   IconNames.CALCULATOR
 ];
 
+export const assessmentTypesToNavlinkInfo = ({
+  assessmentTypes,
+  courseId,
+  isEnrolledInACourse
+}: {
+  assessmentTypes?: string[];
+  courseId?: number;
+  isEnrolledInACourse: boolean;
+}): NavbarEntryInfo[] =>
+  (assessmentTypes || []).map((assessmentType, idx) => ({
+    to: `/courses/${courseId}/${assessmentTypeLink(assessmentType)}`,
+    icon: icons[idx],
+    text: assessmentType,
+    disabled: !isEnrolledInACourse,
+    hasNotifications: true,
+    hiddenInBreakpoints: ['xs', 'sm']
+  }));
+
+export const getStaffNavlinkInfo = ({
+  courseId,
+  role
+}: {
+  courseId?: number;
+  role?: Role;
+}): NavbarEntryInfo[] => {
+  const isStaffOrAdmin = role === Role.Admin || role === Role.Staff;
+  const isAdmin = role === Role.Admin;
+
+  return [
+    {
+      to: `/courses/${courseId}/groundcontrol`,
+      icon: IconNames.SATELLITE,
+      text: 'Ground Control',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm']
+    },
+    {
+      to: `/courses/${courseId}/dashboard`,
+      icon: IconNames.GLOBE,
+      text: 'Dashboard',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm']
+    },
+    {
+      to: `/courses/${courseId}/sourcereel`,
+      icon: IconNames.MOBILE_VIDEO,
+      text: 'Sourcereel',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/grading`,
+      icon: IconNames.ENDORSED,
+      text: 'Grading',
+      disabled: !isStaffOrAdmin,
+      hasNotifications: true,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/storysimulator`,
+      icon: IconNames.CROWN,
+      text: 'Story Simulator',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/xpcalculation`,
+      icon: IconNames.CALCULATOR,
+      text: 'XP Calculation',
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
+    },
+    {
+      to: `/courses/${courseId}/adminpanel`,
+      icon: IconNames.SETTINGS,
+      text: 'Admin Panel',
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
+    }
```

Comment:
Are the ordering changes (originally they were XP, Grading, Story, Admin Panel) intentionally part of the UX improvements?

Reply:
yes. lets keep the admin related routes to the right